### DISCUSSION TITLE
Grouping renovate's created PRs

### DISCUSSION THREAD

**Discussion by bigstinky86** [renovatebot/renovate#11908] [Request Help]
Hello,

We have a bunch of repositories where package.json is stored under the same subfolder. In the same time one of the repositories contains package.json under the root folder. For the repos with package.json under subfolder renovate creates the one PR for package.json updates and for lock file updates but for the repo where package.json is in the root folder it creates the separate PRs for the package.json and for lock file. The configuration below. Could you please clarify what I did wrong?

```
"packageRules": [
      {
        "matchPackagePatterns": ["^@somecompany/*"],
        "matchFiles": ["subfolder/package.json"],
        "matchUpdateTypes": ["major", "minor", "patch", "lockFileMaintenance", "bump"],
        "groupName": "group-libs",
        "commitMessageAction": "Update"
      },
      {
        "excludePackagePatterns": ["^@somecompany"],
        "enabled": false
      },
      {
        "matchSourceUrlPrefixes": ["https://bitbucket.org/somecompany/platform"],
        "matchFiles": ["package.json"],
        "matchPackagePatterns": ["^@somecompany/*"],
        "matchUpdateTypes": ["major", "minor", "patch", "lockFileMaintenance", "bump"],
        "groupName": "group-libs"
      }
    ]
```

To be precise it creates a PR with package.json and package-lock.json files updated and a separate one with package-lock.json file updated. In the first one package-lock.json contains only @somecompany/lib updates and the second one contains all other dependencies updates.

**Comment by rarkins** [comment]
Please clarify your question to more clearly describe:
- What you wanted to happen
- What you tried to configure
- What happened instead

**Reply by bigstinky86** [reply]
- I'd like the only one PR to be created for all the repositories in a list.
- I tried to configure renovate to create the only one PR for the repository where package.json file is stored under the root and the same for repositories where package.json is stored under subfolder
- As I described renovate created one PR with package.json and package-lock.json for the repositories where package.json is stored under subfolder and two PRs (the first with package.json and package-lock.json and the second with package-lock.json changes) for the repo where package.json is stored in the root of the repository.

**Reply by rarkins** [reply]
I don't understand the above, I think because you are using the wrong word repository/repositories when you maybe mean package or directory. Renovate runs on a single repository at a time, there is no such thing as "repositories in a list".

**Reply by bigstinky86** [reply]
Renovate config.js contains `repositories` section as you may know and it is an array of strings. There are specified 10 repos for example. One of these ten repos contains package.json in the root folder. Other ones in the subfolder.

For each of these nine repositories the only one PR is created with lock and package files updated. For the last one repository it creates two PRs. More details about these two PRs I described in my two previous mesages.

**Reply by rarkins** [reply]
Sorry, I tried but I don't understand your description.

**Reply by bigstinky86** [reply]
What actually you didn't get? Is it clear that I run renovate over a couple of repositories and almost for everyone of them it works as I need?
